### ISSUE TITLE
User dashboard badges API endpoint

### ISSUE THREAD

**Issue by stenington** [mozilla/openbadges-discovery#59]
Add an endpoint to get a user's badges from their backpack.

Alternatively I wonder if we want to provide some sort of "I earned this" button that lets the discovery app record that a user has claimed to have a badge **without** requiring them to share it through the Displayer API.

**Comment by threeqube** [comment]
Sorry @stenington I don't follow. Where would the "I earned this" button be?

**Comment by stenington** [comment]
Okay it's clicking what @jpcamara was saying earlier and what I learned last year with the early prototype... the Displayer API is a clunky way to get at someone's earned badges. You have to get all groups, get the badges in each, then manually de-dupe. 

But, it's what we have. I wrote a small client for it at stenington/displayer-api-client although that won't (yet?) do the de-duping.

**Comment by stenington** [comment]
Hmm, in writing code to do the full fetch and de-dupe, I realized there's an additional wrinkle in that the Displayer API returns badges in more or less their underlying format, which can be 0.5 or 1.0 assertions. The directory however will only know about 1.0 AFAIK, because 1.0 is what gives you a separate BadgeClass object. So I suppose somewhere in the client code we should be filtering out 0.5 badges returned through the Issuer API, although then I wonder how to communicate to users the reason why some of their earned and shared badges can't be brought in to the pathways app.

Basically this is bigger than I thought, and gets into some of the "real world" ugliness I'm not entirely sure how to navigate for our first major deadline.

**Comment by threeqube** [comment]
Interesting. cc @brianloveswords and @cmcavoy for good measure.

**Comment by brianloveswords** [comment]
The Displayer API in the current backpack definitely needs an overhaul. (I could have written that sentence “the current backpack needs an overhaul”)
1. I wanna get metrics on how many 0.5 badges are coming in. I don't believe we have `createdOn` dates in the database (I wish we did), but we can look at the last 100 IDs as a proxy, maybe.
2. Also having a hard time thinking of a decent way to communicate to the user why certain badges could not be imported. Maybe something like “Some of the badges in your collection are incompatible with the discovery app. For more information, see this page” and have a page (maybe Sue can help write it?) about the major difference between 0.5 and 1.0 badges and why Discovery requires 1.0 badges.

**Comment by chloeatplay** [comment]
1. @brianloveswords we could have @SueSmith add this to the FAQ for the tool. :+1:

**Comment by SueSmith** [comment]
Sounds good, could add an FAQ on it and the import error/ feedback could link directly to that question on the FAQ page..

**Comment by chloeatplay** [comment]
:+1:

**Comment by chloeatplay** [comment]
Hey @stenington should we close this issue? We have added a section to the FAQ.

**Comment by SueSmith** [comment]
@chloeatplay oh should I have been doing something with this? Had forgotten about this conversation.. :confused:

P.S. am quite happy to have a go at making copy edits etc via pull requests if any text needs changed or added to as we go along..

**Comment by chloeatplay** [comment]
@SueSmith no worries at all :) created a section in the etherpad https://openbadges.etherpad.mozilla.org/discovery-copy . Will add separate tickets for ehemm actually creating those two pages, ABOUT/FAQ. :pray:

**Comment by SueSmith** [comment]
@chloeatplay ah I see it - can draft something for it now if that's cool.

**Comment by chloeatplay** [comment]
:thumbsup:

**Comment by SueSmith** [comment]
Added a short answer https://openbadges.etherpad.mozilla.org/discovery-copy

Totally not sure I'm explaining it correctly (probably because I'm not sure I'm understanding it correctly :neutral_face:).

**Comment by threeqube** [comment]
As far as I can tell, About and FAQ look great to me! :+1:
